client/assets/python/aichat/utils.py

```
@@ -0,0 +1,192 @@
+    if not os.path.exists(local_path) or not os.listdir(local_path):
+        print(f"[LOADER] Local model not found at {local_path}.")
+        
+        # First, try custom archive path if provided
+        if custom_archive_path and handle_local_archive(custom_archive_path, local_path):
+            return True
+        
+        # Check for compressed tar files before downloading
+        tar_path = get_local_zip_path(model_id)  # This returns .tar.gz path
+
+
+        print(f"[LOADER] No local archive found. Starting download from MyData Tools cache server...")
```

Ruff (0.14.5)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)